File: ClassroomMgmt/src/models/StudentHistory.ts
import { StudentTrainingGrade } from "./StudentTrainingGrade";

export interface StudentHistory {
    id: string;
    studentId: string;
    studentTrainingGradeIds: string[];
    studentTrainingGrades: StudentTrainingGrade[];
    progressionId?: string;
}
export const EMPTY_STUDENT_HISTORY: StudentHistory = {
    id: "",
    studentId: "",
    studentTrainingGradeIds: [],
    progressionId: "",
    studentTrainingGrades: [],
    
}

File: ClassroomServer/Models/StudentHistory.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ClassroomServer.Models
{
    public class StudentHistory
    {
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid StudentId { get; set; }

        public List<Guid>? StudentTrainingGradeIds { get; set; } = new();

        public virtual ICollection<StudentTrainingGrade>? StudentTrainingGrades { get; set; } = null;
    }
}